### PR TITLE
[PVR] Store epg database on exit + epg code cleanup

### PR DESCRIPTION
**Pull request by ksooo**
First commit fixes: EPG database changes are saved with a fixed interval of 60 secs. On exit, any pending changes were not written to db.

Second+ commits: Code cleanup along the way.

Runtime-tested on macOS, latest Kodi master.

@Jalle19 do you have some spare time for a code review?

### CODE REVIEW

File: xbmc/pvr/epg/EpgContainer.cpp [hunk 1]
```diff
@@ -765,14 +765,15 @@ const CDateTime CPVREpgContainer::GetLastEPGDate(void)
 {
   CDateTime returnValue;
 
-  CSingleLock lock(m_critSection);
-  for (const auto &epgEntry : m_epgs)
+  m_critSection.lock();
```
Comment (Jalle19) [inline xbmc/pvr/epg/EpgContainer.cpp]: I think most of the files use this convention for locking a certain section only:

```
{
  CSingleLock lock(section);
  /* do work */
}
```

File: xbmc/pvr/epg/EpgContainer.h [hunk 2]
```diff
@@ -206,6 +199,19 @@ namespace PVR
      */
     int GetFutureDaysToDisplay() const;
 
+    /*!
+     * @brief Inform the epg container that playback of an item just started.
+     * @param item The item that started to play.
+     */
+    void OnPlaybackStarted(const CFileItemPtr &item);
```
Comment (Jalle19) [inline xbmc/pvr/epg/EpgContainer.h]: Do we have an interface for these methods somewhere? Seems like a lot of classes have methods like this.

File: xbmc/pvr/epg/EpgContainer.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgContainer.cpp]: Hehe, true and I actually had rewritten this to:

```C++
  std::map<unsigned int, CPVREpgPtr> epgs;
  {
    CSingleLock lock(m_critSection);
    epgs = m_epgs;
  }
````

 but suddenly felt the lock-do-unlock- sequnce more elegant. But I can change this if you like.

File: xbmc/pvr/epg/EpgContainer.h (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgContainer.h]: Good point. We do not currently have such an interface.

Comment (ksooo) [reply xbmc/pvr/epg/EpgContainer.h]: Regarding the naming: how about `IPlaybackStateListener` ? Where to put it file-wise)

File: xbmc/pvr/epg/EpgContainer.cpp (hunk 1 above)
Comment (Jalle19) [reply xbmc/pvr/epg/EpgContainer.cpp]: I don't have a strong preference either way

File: xbmc/pvr/epg/EpgContainer.h (hunk 2 above)
Comment (Jalle19) [reply xbmc/pvr/epg/EpgContainer.h]: The name sounds good, but I'm afraid I don't know off the top of my head where to put such an interface.

**Pull request merged**